pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -1041,5 +1054,22 @@ private ExecutorService getInternalExecutor(Message<T> msg) {
         return executor;
     }
 
+    // If message consumer epoch is smaller than consumer epoch present that
+    // it has been sent to the client before the user calls redeliverUnacknowledgedMessages, this message is invalid.
+    // so we should release this message and receive again
+    protected boolean isValidConsumerEpoch(MessageImpl<T> message) {
```

Comment:
If true means the message is invalid, we should use `isInvalidConsumerEpoch`

Reply:
Or we should return false if the epoch is invalid. I prefer to use `isValidConsumerEpoch` and return false if the epoch is invalid.